snake-kmp-benchmarks/src/jvmMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/benchmark/SnakeyamlEngineJvmLoadingTimeBenchmark.kt

```diff
@@ -0,0 +1,52 @@
+package it.krzeminski.snakeyaml.engine.kmp.benchmark
+
+import kotlinx.benchmark.*
+import okio.Path.Companion.toPath
+import okio.buffer
+import okio.use
+import org.snakeyaml.engine.v2.api.LoadSettings
+import org.snakeyaml.engine.v2.api.YamlUnicodeReader
+import org.snakeyaml.engine.v2.composer.Composer
+import org.snakeyaml.engine.v2.constructor.BaseConstructor
+import org.snakeyaml.engine.v2.constructor.StandardConstructor
+import org.snakeyaml.engine.v2.parser.ParserImpl
+import org.snakeyaml.engine.v2.scanner.StreamReader
+
+@State(Scope.Benchmark)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(BenchmarkTimeUnit.MILLISECONDS)
+class SnakeyamlEngineJvmLoadingTimeBenchmark {
+    @Param("")
+    var openAiYamlPath: String = ""
+
+    private val loadSettings = LoadSettings.builder().build()
+
+    private lateinit var constructor: BaseConstructor
+
+    @Setup
+    fun setUp() {
+        constructor = StandardConstructor(loadSettings)
+    }
+
+    @Benchmark
+    fun loadsOpenAiSchema(): Map<*, *> {
+        return with(fileSystem()) {
+            openReadOnly(openAiYamlPath.toPath(normalize = true)).use { handle ->
+                handle.source().buffer().use { source ->
+                    val reader = StreamReader(
+                        loadSettings,
+                        YamlUnicodeReader(source.inputStream()).readText(),
```

Comment:
note: if I am not mistaken this StreamReader has a constructor that accepts Reader. Maybe use it instead of reading the whole file content into a memory?

Reply:
Good point, not sure why I haven't seen this overload. Changed in https://github.com/krzema12/snakeyaml-engine-kmp/pull/217/commits/0d617995846ff8a793c18a2db7bb025ebfe5e29a